src/org/omegat/core/dictionaries/LingvoDSL.java

```diff
@@ -64,161 +52,226 @@
-                data.add(word.toString(), trans.toString());
+        private String getMediaUrl() {
+            return new File(basePath, previousText).toURI().toString();
+        }
+
+        private boolean isMediaImage() {
+            String ext = FilenameUtils.getExtension(previousText);
+            for (String e : IMAGE_EXTS) {
+                if (e.equals(ext)) {
+                    return true;
+                }
             }
-            data.done();
+            return false;
         }
 
+        /**
+         * Visit an EndTag.
+         *
+         * @param endTag to visit.
+         */
         @Override
-        public List<DictionaryEntry> readArticles(String word) {
-            return data.lookUp(word).stream().map(e -> new DictionaryEntry(e.getKey(), e.getValue()))
-                    .collect(Collectors.toList());
+        public void visit(final DslArticle.EndTag endTag) {
+            if (delayText) {
+                if (previousText == null) {
+                    return;
+                }
+                if (endTag.isTagName("video")) {
+                    sb.append("<a href=\"").append(getMediaUrl()).append("\">").append(previousText).append("</a>");
+                } else if (endTag.isTagName("s")) {
+                    if (isMediaImage()) {
+                        sb.append("<img src=\"").append(getMediaUrl()).append("\" />");
+                    } else {  // sound and unknown files
+                        sb.append("<a href=\"").append(getMediaUrl()).append("\" >").append(previousText).append("</a>");
+                    }
+                } else if (endTag.isTagName("url")) {
+                    sb.append("<a href=\"").append(previousText).append("\">").append(previousText).append("</a>");
+                }
+                delayText = false;
+                previousText = null;
+            }
+            if (endTag.isTagName("b")) {
+                sb.append("</strong>");
+            } else if (endTag.isTagName("u") || endTag.isTagName("i") ||
+                    endTag.isTagName("c") || endTag.isTagName("'")) {
+                sb.append("</span>");
+            } else if (endTag.isTagName("t")) {
+                sb.append("&nbsp;");
+            } else if (endTag.isTagName("sup")) {
+                sb.append("</sup>");
+            } else if (endTag.isTagName("sub")) {
+                sb.append("</sub>");
+            } else if (endTag.isTagName("m")) {
+                sb.append("</p>");
+            }
         }
 
+        /**
+         * Return result.
+         *
+         * @return result.
+         */
         @Override
-        public List<DictionaryEntry> readArticlesPredictive(String word) {
-            return data.lookUpPredictive(word).stream().map(e -> new DictionaryEntry(e.getKey(), e.getValue()))
-                    .collect(Collectors.toList());
+        public String getObject() {
+            if (sb == null) {
+                return "";
```

Comment:
Just an idea: You can make calling `getObject` before `start` an error and throw an exception in that case.